snapcraft/tests/__init__.py

```diff
@@ -26,6 +26,12 @@
 from snapcraft.tests import fixture_setup
 
 
+class ContainsList(list):
+
+        def __eq__(self, other):
+            return all([i[0] in i[0] for i in zip(self, other)])
```

Comment:
I have no idea what this is doing.  Well I can gather from the name what it's trying to do but the `i[0] in i[0]` part seems strange to me.

```
>>> s = ['0', '1', '2']
>>> o = ['1', '2', '3']
>>> all([i[0] in i[0] for i in zip(s, o)])
True
>>> all([i[0] in i[1] for i in zip(s, o)])
False
>>> [i[0] in i[0] for i in zip(s, o)]
[True, True, True]
>>> [(i, i[0] in i[0]) for i in zip(s, o)]
[(('0', '1'), True), (('1', '2'), True), (('2', '3'), True)]
>>> t = ['a', 'b', 'c']
>>> [(i, i[0] in i[0]) for i in zip(s, t)]
[(('0', 'a'), True), (('1', 'b'), True), (('2', 'c'), True)]
```